### PR TITLE
[core][performance] Changed some memcpy calls to use constexpr sizes

### PR DESCRIPTION
**Pull request by rynewang**
In Ray IDs there are some `memcpy`s. Ideally we should always use immediate values for the values call,  because we know it statically. In reality we sometimes used a dynamic one. This PR fixes it.

The perf gain should be minor though.

### CODE REVIEW

File: src/ray/common/id.cc [hunk 1]
```diff
@@ -70,11 +70,10 @@ void FillNil(T *data) {
 }
 
 WorkerID ComputeDriverIdFromJob(const JobID &job_id) {
-  std::vector<uint8_t> data(WorkerID::Size(), 0);
+  std::string data(WorkerID::Size(), '\0');
```
Comment (rkooo567) [inline src/ray/common/id.cc:73]: Q: cpp strings are not immutable right? i.e., fill_N replaces chars in place

File: src/ray/common/id.h [hunk 2]
```diff
@@ -401,10 +403,12 @@ std::ostream &operator<<(std::ostream &os, const PlacementGroupID &id);
                                                                                          \
    private:                                                                              \
     explicit type(const std::string &binary) {                                           \
-      RAY_CHECK(binary.size() == Size() || binary.size() == 0)                           \
-          << "expected size is " << Size() << ", but got data " << binary << " of size " \
-          << binary.size();                                                              \
-      std::memcpy(&id_, binary.data(), binary.size());                                   \
+      if (!binary.empty()) {                                                             \
+        RAY_CHECK(binary.size() == Size())                                               \
```
Comment (rkooo567) [inline src/ray/common/id.h:407]: why removing binary.size() == 0?>

**Review by rkooo567 — APPROVED** [review]

File: src/ray/common/id.cc (hunk 1 above)
Comment (rynewang) [reply src/ray/common/id.cc:73]: Right it’s just like vector of char

File: src/ray/common/id.h (hunk 2 above)
Comment (rynewang) [reply src/ray/common/id.h:407]: Already checked in binary.empty()

**Pull request merged**